include/utils.hpp

```diff
@@ -37,9 +37,113 @@ namespace utils {
      * The cheap way to fix it is to limit rank to 99
     */
 
-    uint64_t r = (current * 100) / total; 
-    if (r > 99) return 99;
-    return r;
+    float rank_coefs[99] = { 0.00,
```

Comment:
can we ensure this is a constant so it doesn't get allocated and deallocated inside here all the time?

Reply:
Roger that, adding `const`.